### ISSUE TITLE
Webview advanced browsing [Kreyul, ♡Sakkera♡, Dog Knight, FrzR, Diskkk]

### ISSUE THREAD

**Issue by RobbWatershed** [avluis/Hentoid#276]
- [x] Add a "forward" button

- [x] Add a "back to gallery" button that goes back in browsing history

**User story for "back to gallery"**
I've been browsing a certain source for 10 minutes and have a valuable history I don't want to lose (e.g. Complex search filter, page number in paged results). When reaching a book, I read 10 pages before deciding it's not a worthy book. At that moment, I need to go back to the gallery or results page to continue searching the results. The current implementation requires to tap back 10 times to achieve that. I need a "back to gallery" button that automatically goes back into history to the latest visited gallery page.
In other words, it's a shortcut to a past point in your history.

~~Add a "cancel download" to the book page~~

~~Rationale : if you accidentally click download and need to cancel it, you need too open the queue and that makes you lose your browsing history~~
=> Moved to #282

- [x] Add a command to copy the current URL or the current "launch code" to the clipboard

- [x] Add an option to remember last position when leaving the webview through the menu, so that reopening the same source doesn't go back to the main page

**Comment by AlfredPaguio** [comment]
![IMG_20190802_004654](https://user-images.githubusercontent.com/25319699/62368168-635b6c80-b55e-11e9-8fdd-4cb669e42efa.jpg)
Possible solution for :
 "Add a "cancel download" to the book page
Rationale : if you accidentally click download and need to cancel it, you need too open the queue and that makes you lose your browing history

  Add option to remember last position when leaving the webview through the menu, so that reopening the same source doesn't go back to the main page

  Add a command to copy the current URL or the current "launch code" to the clipboard"

**Comment by RobbWatershed** [comment]
Last viewed page is restored but browsing history cannot be restored :
- no write equivalent to `Webview.copyBackForwardList()` 
- webview is a mock class that wraps around either the system web view or Chrome. So no chance to interfere with its inner workings by subclassing it
- the last option would be to implement a manual browsing history management, but that's super heavy
=> Last page is restored without the browsing history
